src/systemctl/systemctl.c

```diff
@@ -2524,6 +2537,10 @@ static int unit_find_paths(
                                         return r;
                         }
                 }
+                else if (!endswith(path, unit_name)) {
+                        set_remove(names, unit_name);
+                        set_put(names, basename(path));
```

Comment:
hmm, also, why add the original unit_name to the set anyway, if we drop it here right-away again? please simple drop the earlier code for adding the name to the set and just do it here, but unconditionally.